tests/integration/targets/ndo_interface_setting/tasks/main.yml

```diff
@@ -0,0 +1,721 @@
+          - nm_add_interface_policy_group_again.previous.linkLevel.autoNegotiation == nm_add_interface_policy_group_again.current.linkLevel.autoNegotiation == "on"
+          - nm_add_interface_policy_group_again.previous.linkLevel.bringUpDelay == nm_add_interface_policy_group_again.current.linkLevel.bringUpDelay == 0
+          - nm_add_interface_policy_group_again.previous.linkLevel.debounceInterval == nm_add_interface_policy_group_again.current.linkLevel.debounceInterval == 100
+          - nm_add_interface_policy_group_again.previous.linkLevel.fec == nm_add_interface_policy_group_again.current.linkLevel.fec == "inherit"
+          - nm_add_interface_policy_group_again.previous.mcp.adminState == nm_add_interface_policy_group_again.current.mcp.adminState == "enabled"
+          - nm_add_interface_policy_group_again.previous.mcp.mcpMode == nm_add_interface_policy_group_again.current.mcp.mcpMode == "off"
+          - nm_add_interface_policy_group_again.previous.mcp.initialDelayTime == nm_add_interface_policy_group_again.current.mcp.initialDelayTime == 0
+          - nm_add_interface_policy_group_again.previous.mcp.txFreq == nm_add_interface_policy_group_again.current.mcp.txFreq == 0
+          - nm_add_interface_policy_group_again.previous.mcp.txFreqMsec == nm_add_interface_policy_group_again.current.mcp.txFreqMsec == 500
+          - nm_add_interface_policy_group_again.previous.mcp.gracePeriod == nm_add_interface_policy_group_again.current.mcp.gracePeriod == 3
+          - nm_add_interface_policy_group_again.previous.mcp.gracePeriodMsec == nm_add_interface_policy_group_again.current.mcp.gracePeriodMsec == 0
+          - nm_add_interface_policy_group_again.previous.portChannelPolicy.minLinks == nm_add_interface_policy_group_again.current.portChannelPolicy.minLinks == 0
+          - nm_add_interface_policy_group_again.previous.portChannelPolicy.maxLinks == nm_add_interface_policy_group_again.current.portChannelPolicy.maxLinks == 0
+          - nm_add_interface_policy_group_again.previous.portChannelPolicy.hashFields == nm_add_interface_policy_group_again.current.portChannelPolicy.hashFields == None
+          - nm_add_interface_policy_group_again.previous.uuid is defined
+          - nm_add_interface_policy_group_again.current.uuid is defined
+
+
+    # Interface Policy group interface_type PC/VPC
+    - name: Create a Interface Policy group of interface_type 'port_channel' (check mode)
+      cisco.mso.ndo_interface_setting: &add_interface_policy_group_pc
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        interface_policy_group: ansible_interface_policy_group_pc
+        interface_type: port_channel
+        speed: 100G
+        auto_negotiation: on_enforce
+        vlan_scope: port_local
+        cdp_admin_state: enabled
+        port_channel_mode: lacp_active
+        min_links: 5
+        max_links: 50
+        control: ["fast_sel_hot_stdby", "graceful_conv", "susp_individual", "symmetric_hash"]
```

Comment:
if we do input sorting, should we not also test assert the order we expect them to be in?

Reply:
doesn't have to be a sorted list. changed the cod ein module